lib/asciidoctor-diagram/extensions.rb

```diff
@@ -108,12 +108,19 @@ def process(parent, reader_or_target, attributes)
           raise "#{self.class.name} does not support output format #{format}" unless generator_info
 
           begin
+            title = source.attributes.delete 'title'
+            caption = source.attributes.delete 'caption'
+
             case generator_info[:type]
               when :literal
-                create_literal_block(parent, source, generator_info)
+                block = create_literal_block(parent, source, generator_info)
               else
-                create_image_block(parent, source, format, generator_info)
+                block = create_image_block(parent, source, format, generator_info)
             end
+
+            block.title = title
+            block.assign_caption(caption, 'figure')
```

Comment:
You'll need to add a bit more logic to cover the functionality that was lost.

See https://github.com/asciidoctor/asciidoctor/blob/v1.5.5/lib/asciidoctor/parser.rb#L909-L919

Reply:
I've added registration of the image in the catalog, the scaledwidth fix and special character substitution on the alt text.

Reply:
There's actually slightly newer code for this, which may actually be easier to maintain.

https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/parser.rb#L545-L551

I'd import the regular expression into the class since those are not part of the public API.

Reply:
I think the character range test is fine to be honest. It'll also be faster than evaluating a regex. In particular in a Unicode world `\d` is a more expensive test than `[0-9]`. Since the width will always use ASCII digit characters, `[0-9]` is correct. And since you're only interested in the last character, testing just that seems fine to me.